### PR TITLE
Added install folder caching in GitHub Actions for Windows and MacOS runs

### PR DESCRIPTION
**Pull request by mmzakhar**

### CODE REVIEW

File: scripts/build_windows.bat [hunk 1]
```diff
@@ -13,9 +13,9 @@ IF "%VS_VER%"=="2019_build_tools" (
 @call "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build\vcvars64.bat"
 )
 
-@call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"
+@call "C:\Program Files (x86)\Intel\oneAPI\compiler\2021.1-beta10\env\vars.bat"
```
Comment (rscohn2) [inline scripts/build_windows.bat:16]: I have had bad luck on linux trying to use the compiler vars.bat instead of the general setvars.bat. It did not put the runtime libraries on the path and it did not put dpc_common on the include path. What is the motivation for this change.?

**Review by rscohn2 — APPROVED** [review]

File: scripts/build_windows.bat (hunk 1 above)
Comment (mmzakhar) [reply scripts/build_windows.bat:16]: - By default, setvars.bat is setting up env. for 'latest' versions of components. On Windows, 'latest' is a Windows symlink - and GitHub Action doesn't cache such filesystem entities. So if the entire oneAPI directory is cached, and setvars.bat is called, no env would be setup because there are no 'latest' folders. There are 2 options to setup env. under caching: call the individual vars scripts directly, or setup a setvars config. Both require explicit specification of versions. I'm not sure which option is better.

- Another reason is that caching entire oneAPI folder leads to large cache sizes (e.g, conda-channel, debugger, FPGA emulator are installed with compiler), so selective caching is a better approach given the current component layout. It makes CI config file larger though. To cache setvars, the CI config would need to grow a bit further (setvars is not a standalone script, it uses other scripts)

Comment (rscohn2) [reply scripts/build_windows.bat:16]: Makes sense. I will be more vigilant in reporting issues with running the vars.bat script.
